### PR TITLE
Docs: add discovery_group to teleport.yaml/discovery_service examples

### PR DESCRIPTION
**Pull request by marcoandredinis**
When the discovery_group is not configured, teleport will log a warning message saying that it is recommended.
https://github.com/gravitational/teleport/blob/24e8b68f3aa29074354e95b3cfec5938a81ea71e/lib/srv/discovery/discovery.go#L251
Some configuration examples do not include it, so when users use that example to implement their own variation, it will output a warning message.
After seeing that warning users might wonder if there's anything wrong with their `teleport.yaml`.
Instead, we add a discovery_group to all examples so that users don't get that message.

### CODE REVIEW

File: docs/pages/enroll-resources/auto-discovery/kubernetes/google-cloud.mdx [hunk 1]
```diff
@@ -396,6 +396,7 @@ ssh_service:
   enabled: off
 discovery_service:
   enabled: "yes"
+  discovery_group: "gke-myproject"
```
Comment (ptgott) [inline docs/pages/enroll-resources/auto-discovery/kubernetes/google-cloud.mdx:399]: Do we have any instructions telling the user to update `discovery_group` in this and other Teleport configuration examples?

Comment (marcoandredinis) [reply docs/pages/enroll-resources/auto-discovery/kubernetes/google-cloud.mdx:399]: We have it in the reference page and some other configurations.
I can add it to everything.
The description we use in the reference page is:
```yaml
    # discovery_group is used to group discovered resources into different
    # sets. This is required when you have multiple Teleport Discovery services
    # running. It prevents discovered services from colliding in Teleport when
    # managing discovered resources.
    # If two Discovery Services match the same resources, they must be in the
    # same discovery group.
    # If two Discovery Services match different resources, they must be in
    # different discovery groups.
    discovery_group: "disc-group"
```

Comment (ptgott) [reply docs/pages/enroll-resources/auto-discovery/kubernetes/google-cloud.mdx:399]: 👍 I would just make it explicit how to assign this value. Is this the name of a Google Cloud project where Kubernetes clusters you want to discover are running?

Comment (marcoandredinis) [reply docs/pages/enroll-resources/auto-discovery/kubernetes/google-cloud.mdx:399]: It's an opaque value. Can be a random string.
When multiple DiscoveryServices are running, it is used to ensure resources are not deleted when reconciling the list of databases.

Comment (ptgott) [reply docs/pages/enroll-resources/auto-discovery/kubernetes/google-cloud.mdx:415]: 👍 I hadn't realized when leaving this comment that there is a partial, `docs/pages/includes/discovery/discovery-group.mdx`, that provides this information already. I think we can remove the `discovery_group` comments and use this partial instead.

Comment (marcoandredinis) [reply docs/pages/enroll-resources/auto-discovery/kubernetes/google-cloud.mdx:399]: Removed the inline comment in the `yaml` code blocks.
Added the partial where it was missing.

**Review by r0mant — DISMISSED** [review]

File: docs/pages/enroll-resources/auto-discovery/databases/databases.mdx [hunk 2]
```diff
@@ -75,14 +75,6 @@ Here's an example database discovery configuration for the Discovery Service:
...
 discovery_service:
     enabled: true
-    # discovery_group is used to group discovered resources into different
-    # sets. This is required when you have multiple Teleport Discovery services
-    # running. It prevents discovered services from colliding in Teleport when
-    # managing discovered resources.
-    # If two Discovery Services match the same resources, they must be in the
-    # same discovery group.
-    # If two Discovery Services match different resources, they must be in
-    # different discovery groups.
```
Comment (r0mant) [inline docs/pages/enroll-resources/auto-discovery/databases/databases.mdx:85]: Why did you remove these explanations from everywhere? They seemed useful, discovery group is not an obvious concept to understand without the comment.

Comment (marcoandredinis) [reply docs/pages/enroll-resources/auto-discovery/databases/databases.mdx:85]: I removed them but added the `
(!docs/pages/includes/discovery/discovery-group.mdx!)` partial where it was not already present.

I'm fine with either, but I think we should stick to only one form of documentation (either inline with the `yaml` code block or with the partial).

**Review by ptgott — APPROVED** [review]

**Review by r0mant — APPROVED** [review]
